at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -491,28 +494,37 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
         var metaData =
             await _atClient.getLocalSecondary()!.keyStore!.getMeta(key);
         if (metaData != null) {
-          key = '$key$_metadataToString(metaData)';
+          key = '$key${_metadataToString(metaData)}';
         }
         command = 'update:meta:$key';
         break;
       case CommitOp.UPDATE_ALL:
         var key = entry.atKey;
-        var value = await _atClient.getLocalSecondary()!.keyStore!.get(key);
-        var metaData =
-            await _atClient.getLocalSecondary()!.keyStore!.getMeta(key);
+        AtData value = await _atClient.getLocalSecondary()!.keyStore!.get(key);
         var keyGen = '';
-        if (metaData != null) {
-          keyGen = _metadataToString(metaData);
+        // If metadata.encoding is not empty, the value is encoded.
+        // When value is encoded, use update:json variant in update regex to
+        // send command to cloud secondary
```

Comment:
I do like this approach in the client. However, we need to ensure that the non-json update verb syntax also supports `encoding`

Reply:
Removed the JSON approach and made changes to use to update regex.